syft/frameworks/torch/tensors/interpreters/replicated_shared.py

```diff
@@ -1,39 +1,54 @@
-import random
-from operator import add, sub
+from operator import add, sub, mul
 import torch
 import syft
 from syft.generic.abstract.tensor import AbstractTensor
+from syft.frameworks.torch.mpc.przs import PRZS, gen_alpha_3of3
 
 
 class ReplicatedSharingTensor(AbstractTensor):
     def __init__(
-        self, shares_map=None, owner=None, id=None, tags=None, description=None,
+        self, plain_text=None, players=None, owner=None, id=None, tags=None, description=None,
     ):
         super().__init__(id=id, owner=owner, tags=tags, description=description)
+        self.ring_size = 2 ** 32
+        shares_map = self.__generate_shares_map(plain_text, players)
         self.child = shares_map
-        self.ring_size = 2 ** 5
 
-    def share_secret(self, secret, workers):
+    def __generate_shares_map(self, plain_text, players):
+        if plain_text is None or players is None:
+            return None
+        elif isinstance(plain_text, torch.Tensor):
+            return self.share_secret(plain_text, players)
+        elif plain_text is ReplicatedSharingTensor:
+            return plain_text.get_shares_map()
+        else:
+            raise ValueError(f" {type(plain_text)} is not supported")
+
+    def share_secret(self, plain_text, workers):
         number_of_shares = len(workers)
         workers = self.__arrange_workers(list(workers))
-        shares = self.generate_shares(secret, number_of_shares)
+        shares = self.generate_shares(plain_text, number_of_shares)
         shares_map = self.__distribute_shares(workers, shares)
-        self.child = shares_map
-        return self
+        PRZS.setup(workers)
+        return shares_map
 
     @staticmethod
     def __arrange_workers(workers):
         """ having local worker in index 0 saves one communication round"""
-        if syft.hook.local_worker in workers:
-            workers.remove(syft.hook.local_worker)
-            workers = [syft.hook.local_worker] + workers
+        if len(workers) != 3:
+            raise ValueError("you must provide 3 players")
+        me = syft.hook.local_worker
+        if me in workers:
+            workers.remove(me)
+            workers = [me] + workers
         return workers
 
     def generate_shares(self, plain_text, number_of_shares=3):
         shares = []
+        plain_text.long()
```

Comment:
Q: In case plain_text is a ```float``` this will simply do a "cast" to an ```int``` type, right?

Reply:
it will cast it from any torch.dtype to torch.long